### ISSUE TITLE
[RFE] Uniform output of socket-collector and process-collector when no container matches the requested filter

### ISSUE THREAD

**Issue by blanquicet** [kinvolk/inspektor-gadget#253]
## Current situation

As discussed in https://github.com/kinvolk/inspektor-gadget/pull/247#discussion_r701038950, when no container matches the requested filter, the process-collector show an empty list: 
```
$ kubectl gadget process-collector -A -l k8s-app=non-existent
NAMESPACE    POD    CONTAINER    COMM    PID
```

While the socket-collector report an error:
```
$ ./kubectl-gadget-linux-amd64 socket-collector -A -l k8s-app=non-existent
WARN[0002] Error getting traces from node "my-node": No container matches the requested filter: {Namespace: Podname: Labels:map[k8s-app:non-existent] ContainerName:}  args="[]" command="kubectl-gadget socket-collector"
FATA[0002] Error getting traces from all nodes           args="[]" command="kubectl-gadget socket-collector"
```

## Ideal future situation

Gadgets could be uniformed taking into account that output must not lead into a misunderstood.

**Comment by blanquicet** [comment]
Following my comments after discussion in https://github.com/kinvolk/inspektor-gadget/pull/299#issuecomment-923131707:

To improve and align the output of socket-collector and process-collector when a given namespace or pod does not exist, I propose to check if the requested namespace and pod exist before creating the traces in `GenericTraceCommand()`. In this way, this scenario is managed in a shared/common place for all the gadgets that capture a snapshot (No stream). I know that I am anticipating the check and giving less time to pod/namespace to be created but I think that the user should ensure (Maybe using `kubectl wait`) that namespace/pod is ready before launching the gadget. What do you think?

**Comment by mauriciovasquezbernal** [comment]
>  I know that I am anticipating the check and giving less time to pod/namespace to be created but I think that the user should ensure (Maybe using kubectl wait) that namespace/pod is ready before launching the gadget. What do you think?

I don't think it makes any difference. The namespace / pod should exist before the user launches the command.
